cli/tools/upgrade.rs

```diff
@@ -623,65 +627,30 @@ fn select_specific_version_for_upgrade(
   version: String,
   force: bool,
 ) -> Result<Option<AvailableVersion>, AnyError> {
-  match release_channel {
-    ReleaseChannel::Stable => {
-      let current_is_passed = if version::DENO_VERSION_INFO.release_channel
-        != ReleaseChannel::Canary
-      {
-        version::DENO_VERSION_INFO.deno == version
-      } else {
-        false
-      };
-
-      if !force && current_is_passed {
-        log::info!(
-          "Version {} is already installed",
-          version::DENO_VERSION_INFO.deno
-        );
-        return Ok(None);
-      }
-
-      Ok(Some(AvailableVersion {
-        version_or_hash: version,
-        release_channel,
-      }))
-    }
-    ReleaseChannel::Canary => {
-      let current_is_passed = version::DENO_VERSION_INFO.git_hash == version;
-      if !force && current_is_passed {
-        log::info!(
-          "Version {} is already installed",
-          version::DENO_VERSION_INFO.deno
-        );
-        return Ok(None);
-      }
-
-      Ok(Some(AvailableVersion {
-        version_or_hash: version,
-        release_channel,
-      }))
+  let current_is_passed = match release_channel {
+    ReleaseChannel::Stable | ReleaseChannel::Lts => {
+      version::DENO_VERSION_INFO.release_channel != ReleaseChannel::Canary
+        && version::DENO_VERSION_INFO.deno == version
     }
+    ReleaseChannel::Canary => version::DENO_VERSION_INFO.git_hash == version,
     ReleaseChannel::Rc => {
-      let current_is_passed = version::DENO_VERSION_INFO.release_channel
-        == ReleaseChannel::Rc
-        && version::DENO_VERSION_INFO.deno == version;
-
-      if !force && current_is_passed {
-        log::info!(
-          "Version {} is already installed",
-          version::DENO_VERSION_INFO.deno
-        );
-        return Ok(None);
-      }
-
-      Ok(Some(AvailableVersion {
-        version_or_hash: version,
-        release_channel,
-      }))
+      version::DENO_VERSION_INFO.release_channel == ReleaseChannel::Rc
+        && version::DENO_VERSION_INFO.deno == version
```

Comment:
Now it occurred to me that we can use the same check for all release channels, but canary - we want to check if the current binary's release channel is the one requested and then compare if `version::DENO_VERSION_INFO.deno` is the same

Reply:
Done.